### PR TITLE
LOK-2254: WIP: Add trace ID to graphQL errors shown in the UI

### PR DESCRIPTION
**Pull request by deejgregor**
This is a work-in-progress and is hacky and definitely needs cleaned up.

## Description

Add trace ID (if available) to errors shown in the UI.

Our web server adds a `Server-Timing` header on all responses (including errors) with the traceparent header. This pulls out the trace ID portion of that and adds it to errors shown in the UI as a snackbar (the red popup at the bottom center of the screen).

Here's what the header looks lke, BTW:
```
$ curl --dump-header /dev/stdout -sSf https://auth.onms-fb-dev.dev.nonprod.dataservice.opennms.com/ | grep -i Server-timing
curl: (22) The requested URL returned error: 404
server-timing: traceparent;desc="00-ee7c5ef3834410c3f7a74ebf2146f18c-830c100f808d18f6-01"
```

There a few issues I ran into while doing this (I'm not sure if any of these need to be addressed as part of this PR, I just wanted to mention them as an FYI because they definitely impacted my attempts to do this, in case anyone else is playing around):
1. In the UI development mode, the `Server-Timing` header is never visible in the JavaScript code, even with `Timing-Allow-Origin:  *`. I've found some references to browsers not making this visible when there is a mix of HTTP and HTTPS.
2. Some of the responses don't have the `response` object. I haven't dug into why that happens in villus.
3. When we get multiple errors in short succession, I think each one of them will get their own snackbar and they'll cover each other up, only showing the most recent. (the combination of this and previous item confused the heck out of me for awhile)

Super nice-to-have: if it is possible to add links to the content in the snackbar, we could link directly to the trace in Jaeger in local dev environments (making this feature optional and only enabled where we want it), which would make debugging exceptionally easy.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2254

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/src/services/gqlService.ts [hunk 1]
```diff
@@ -21,7 +21,30 @@ const getGqlClient = (kc: Keycloak) => {
 // error notification plugin for specific status codes
 const errorNotificationPlugin = definePlugin(({ afterQuery }) => {
   afterQuery(({ error }, { response }) => {
+    let traceId = ""
```
Comment (synqotik) [inline ui/src/services/gqlService.ts:24]: Consider replacing here through `errorsMsgs` with:

```javascript
if (!error || !response) {
  return
}

const serverTiming = response.headers.get('server-timing')

let traceId = ''

if (serverTiming) {
  const pattern = /traceparent;desc="(..)-([^"-]*)-([^"-]*)-(..)"/
  const result = pattern.exec(serverTiming)

  if (result && result.length > 1) {
    traceId = `Trace ID: ${result[2]}`
  }
}
```

File: ui/src/services/gqlService.ts [hunk 2]
```diff
@@ -21,7 +21,30 @@ const getGqlClient = (kc: Keycloak) => {
...
+//      console.log('token: ', response?.headers.get('token'))
+//      for (const pair of response.headers.entries()) {
+//        console.log(`${pair[0]}: ${pair[1]}`);
+//      }
+    }
     if (!error) return
+    if (!response) return
 
     const errorsMsgs = [
       {
```
Comment (synqotik) [inline ui/src/services/gqlService.ts:50]: Fix `occured` -> `occurred` in a couple cases below.

File: ui/src/services/gqlService.ts [hunk 3]
```diff
@@ -65,17 +88,18 @@ const errorNotificationPlugin = definePlugin(({ afterQuery }) => {
     }
 
     const notificationMsg = () => {
```
Comment (synqotik) [inline ui/src/services/gqlService.ts:90]: Consider replacing this with:

```javascript
const notificationMsg = () => {
  const errorObj = errorsMsgs.filter({ err }) => hasError(err))?.[0]

  return (errorObj?.msg || 'An unknown error has occurred') + ' ' + traceId
}
```

File: ui/src/services/gqlService.ts (hunk 1 above)
Comment (deejgregor) [reply ui/src/services/gqlService.ts]: Done!

File: ui/src/services/gqlService.ts (hunk 3 above)
Comment (deejgregor) [reply ui/src/services/gqlService.ts:84]: Ooh, much nicer. Thank you! Done!

File: ui/src/services/gqlService.ts (hunk 2 above)
Comment (deejgregor) [reply ui/src/services/gqlService.ts:44]: Done. Thanks for catching this.

**Review by synqotik — APPROVED** [review]
Looks good after changes!

**Pull request merged**